lib/commands/context.js

```diff
@@ -299,13 +299,13 @@ extensions.onPageChange = async function (pageChangeNotification) {
       let needsPageLoad = (() => {
         let item = (arr) => {
           return _.filter(arr, (obj) => {
-            return obj.id === this.curContext;
```

Comment:
@cschabetsberger-micro-focus I think this change makes sense. However I'm a bit stuck trying to understand the existing code. With your changes, how I read this section:

```js
        let item = (arr) => {
          return _.filter(arr, (obj) => {
            return obj === this.curContext;
          })[0];
        };
```

Is that it will either result in `this.curContext` or `undefined`, based on whether the array passed in contains something equivalent to `this.curContext`.

So, calling `item(this.contexts)` below seems like a roundabout way to just reference `this.curContext`, no? Surely `this.contexts` would always contain `this.curContext`?

In any case, if I'm reading this right, we're basically just checking whether the new pageArray contains `this.curContext`, and returning `false` if it does, to say we don't need a page change. If that's the case, sounds like we can get rid of that block I mentioned in favor of something more transparent?

@imurchie should probably weigh in on this.